packages/core/src/conflict-resolution.ts

```diff
@@ -271,24 +267,21 @@ export class ConflictResolution {
     const itr = log.entries()
     let entry = itr.next()
     while (!entry.done) {
-      const logEntry = entry.value[1]
-      const commit = await this.dispatcher.retrieveCommit(logEntry.cid)
+      const logEntry = await this.expandLogEntry(entry.value[1])
+      const commitMeta = { cid: logEntry.cid, timestamp: logEntry.timestamp }
       // TODO - should catch potential thrown error here
 
-      let payload = commit
-      if (StreamUtils.isSignedCommit(commit)) {
-        payload = await this.dispatcher.retrieveCommit(commit.link)
-      }
-
-      if (payload.proof) {
-        // it's an anchor commit
+      if (logEntry.commit.proof) {
+        // It's an anchor commit
         // TODO: Anchor validation should be done by the StreamHandler as part of applying the anchor commit
-        await verifyAnchorCommit(this.dispatcher, this.anchorValidator, commit)
-        state = await handler.applyCommit(commit, logEntry, this.context, state)
+        await verifyAnchorCommit(this.dispatcher, this.anchorValidator, logEntry.commit)
+        state = await handler.applyCommit(logEntry.commit, commitMeta, this.context, state)
       } else {
-        // it's a signed commit
-        const tmpState = await handler.applyCommit(commit, logEntry, this.context, state)
-        const isGenesis = !payload.prev
+        // It's a signed commit but may not be using DagJWS. If the JWS envelope is present, use that while applying the
+        // commit, otherwise use the commit itself (e.g. for CAIP-10 links).
+        const commitToApply = logEntry.envelope ? logEntry.envelope : logEntry.commit
+        const tmpState = await handler.applyCommit(commitToApply, commitMeta, this.context, state)
```

Comment:
Don't we need to pass both the envelope and commit here?

Reply:
Eventually, yes. Since I've left the IPFS lookup inside `applyCommit` in there, passing the envelope is sufficient for the moment. In a subsequent update, if I just pass the `LogEntry` itself, I'll be able to remove the internal lookup as well.

Reply:
ok, fair enough :)